### PR TITLE
chore(dotnet): Migrate IBrowserContext

### LINKED ISSUE #1232: Use generated IBrowserContext

### PR DESCRIPTION
**Pull request by avodovnik**
Should close #1232 when done.

### CODE REVIEW

File: src/PlaywrightSharp.Tests/BrowserContextAddCookiesTests.cs [hunk 1]
```diff
@@ -44,7 +44,17 @@ public async Task ShouldRoundtripCookie()
...
+            {
+                Domain = c.Domain,
+                Expires = c.Expires,
+                HttpOnly = c.HttpOnly,
+                Name = c.Name,
+                Path = c.Path,
+                SameSite = c.SameSite,
+                Secure = c.Secure,
+                Value = c.Value
+            }));
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/BrowserContextAddCookiesTests.cs:57]: Can't we make an implicit conversion?

File: src/PlaywrightSharp.Tests/BrowserContextHttpCredentialsTests.cs [hunk 2]
```diff
@@ -26,24 +26,6 @@ public async Task ShouldFailWithoutCredentials()
...
-            var page = await context.NewPageAsync();
-            var response = await page.GoToAsync(TestConstants.EmptyPage);
-            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
-            await context.SetHttpCredentialsAsync(new HttpCredentials
-            {
-                Username = "user",
-                Password = "pass"
-            });
-            response = await page.ReloadAsync();
-            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/BrowserContextHttpCredentialsTests.cs:44]: Can we just comment the code so the tool doesn't detect this as a missing test?

File: src/PlaywrightSharp.Tests/PermissionsTests.cs [hunk 3]
```diff
@@ -153,7 +153,7 @@ public async Task ShouldIsolatePermissionsBetweenBrowserContexts()
             Assert.Equal("prompt", await GetPermissionAsync(otherPage, "geolocation"));
 
             await Context.GrantPermissionsAsync(Array.Empty<string>(), TestConstants.EmptyPage);
-            await otherContext.GrantPermissionsAsync(ContextPermissions.Geolocation, TestConstants.EmptyPage);
+            await otherContext.GrantPermissionsAsync(new string[] { ContextPermissions.Geolocation }, TestConstants.EmptyPage);
```
Comment (kblok) [inline src/PlaywrightSharp.Tests/PermissionsTests.cs:156]:
```suggestion
            await otherContext.GrantPermissionsAsync(new[] { ContextPermissions.Geolocation }, TestConstants.EmptyPage);

```

**Comment by kblok** [comment]
I think we should delete and ignore `src/PlaywrightSharp.Tests/Screenshots/FIREFOX/test.png`

File: src/PlaywrightSharp/ContextEvent.cs [hunk 4]
```diff
@@ -4,19 +4,22 @@
 namespace PlaywrightSharp
 {
     /// <summary>
-    /// Context events. See <see cref="IBrowserContext.WaitForEventAsync{T}(PlaywrightEvent{T}, Func{T, bool}, int?)"/>.
+    /// Context events. See <see cref="IBrowserContext.WaitForEventAsync(string, float?)"/>.
```
Comment (kblok) [inline src/PlaywrightSharp/ContextEvent.cs:7]: I think we could still point to the other function.

File: src/PlaywrightSharp/PlaywrightSharp.csproj [hunk 5]
```diff
@@ -54,11 +54,27 @@
...
     </DriverFiles>
   </ItemGroup>
+  <ItemGroup>
+    <Compile Remove="BrowserContext.old.cs" />
+    <Compile Remove="__TRASH\BrowserContext.old.cs" />
+    <Compile Remove="__TRASH\IBrowserContext.cs" />
+    <Compile Remove="__TRASH\NetworkCookie.cs" />
+    <Compile Remove="__TRASH\SetNetworkCookieParam.cs" />
+    <Compile Remove="__TRASH\TimeoutSettings.cs" />
+  </ItemGroup>
```
Comment (kblok) [inline src/PlaywrightSharp/PlaywrightSharp.csproj:64]: we don't need this

File: src/PlaywrightSharp/PlaywrightSharp.csproj [hunk 6]
```diff
@@ -54,11 +54,27 @@
...
     </ContentWithTargetPath>
   </ItemGroup>
+  <ItemGroup>
+    <None Include="__TRASH\BrowserContext.old.cs" />
+    <None Include="__TRASH\BrowserContext.old.cs" />
+    <None Include="__TRASH\IBrowserContext.cs" />
+    <None Include="__TRASH\NetworkCookie.cs" />
+    <None Include="__TRASH\SetNetworkCookieParam.cs" />
+    <None Include="__TRASH\TimeoutSettings.cs" />
+  </ItemGroup>
```
Comment (kblok) [inline src/PlaywrightSharp/PlaywrightSharp.csproj:77]: we don't need this

File: src/PlaywrightSharp/__TRASH/BrowserContext.old.cs [hunk 7]
```diff
@@ -0,0 +1,444 @@
+using System;
```
Comment (kblok) [inline src/PlaywrightSharp/__TRASH/BrowserContext.old.cs:1]: We don't need this

File: src/PlaywrightSharp/PlaywrightSharp.csproj (hunk 5 above)
Comment (avodovnik) [reply src/PlaywrightSharp/PlaywrightSharp.csproj:64]: BLARGH, I was so focused on getting this through the door I didn't clean it up. Thanks for reminding me!

File: src/PlaywrightSharp.Tests/BrowserContextAddCookiesTests.cs (hunk 1 above)
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/BrowserContextAddCookiesTests.cs:57]: I've removed `SetNetworkCookieParam` class, and would rather keep it explicit for the user to **know** there is a change. It's not _that_ painful; should just be a rename in the code.

File: src/PlaywrightSharp/ContextEvent.cs (hunk 4 above)
Comment (avodovnik) [reply src/PlaywrightSharp/ContextEvent.cs:7]: Yes, now we can, but I've removed it before in the attempt to get rid of the awesomeness that is `PlaywrightEvent<>` :)

**Comment by avodovnik** [comment]
I'm starting to see some more failing tests, looks like there's a bit more work to be done.

**Review by kblok — APPROVED** [review]

**Comment by avodovnik** [comment]
@kblok I've verified both failing tests work on Windows/Chromium locally, so I'm sure it's just flakiness.

**Pull request merged**